codec-http/src/main/java/io/netty/handler/codec/http/cookie/ClientCookieEncoder.java

```diff
@@ -91,12 +121,66 @@ public String encode(Cookie... cookies) {
         }
 
         StringBuilder buf = stringBuilder();
-        for (Cookie c : cookies) {
-            if (c == null) {
-                break;
+        if (strict) {
+            ArrayList<Cookie> cookiesList = new ArrayList<Cookie>(cookies.length);
```

Comment:
Oh, you're right, thanks. I always assumed that you couldn't sort a List, since it doesn't have guaranteed O(1) access time for get(). Makes sense that it makes an interim copy to get around this issue.

However, I don't understand the purpose of the following lines -- how important is it to be able to terminate the cookie list by including a null cookie?

```java
    if (c == null) {
        break;
    }
```

The reason I ask is because to make the change to use an array for sorting, rather than a list, I need to allocate an array and then copy the objects in from the list (or collection or iterable). However, since the list can be terminated by a single null object, I would need to resize the array if a null object is encountered in the list. Can I safely remove the above lines from each of the methods? (The behavior of taking no further cookies after a null cookie is encountered is not documented in the method comments anyway.)